src/tests/conformance/status.rs

```diff
@@ -0,0 +1,85 @@
+use std::time::Duration;
+
+use tempfile::TempDir;
+use tokio::time::{error::Elapsed, sleep, timeout};
+
+use crate::{
+    protocol::{
+        codecs::binary::{BinaryMessage, Payload},
+        proto::TmStatusChange,
+    },
+    setup::node::build_stateful_builder,
+    tools::{
+        config::TestConfig,
+        constants::EXPECTED_RESULT_TIMEOUT,
+        rpc::{get_ledger_info, LedgerInfoResponse, RpcResponse},
+        synth_node::SyntheticNode,
+    },
+};
+
+#[tokio::test]
+#[allow(non_snake_case)]
+async fn c010_TM_STATUS_CHANGE_node_should_send_ledger_information_using_status_change() {
+    let target = TempDir::new().expect("unable to create TempDir");
+
+    // Create a stateful node.
+    let mut node = build_stateful_builder(target.path().to_path_buf())
+        .expect("unable to get stateful builder")
+        .start(false)
+        .await
+        .expect("unable to start stateful node");
+
+    // Connect synth node.
+    let mut sn = SyntheticNode::new(&TestConfig::default()).await;
+    sn.connect(node.addr()).await.unwrap();
+
+    // Get ledger information via RPC.
+    let info = wait_for_ledger_info_by_rpc()
+        .await
+        .expect("no ledger info within required time");
+    let rpc_ledger_index = info
+        .result
+        .ledger
+        .ledger_index
+        .parse::<u32>()
+        .unwrap_or_else(|_| {
+            panic!(
+                "unable to parse ledger_index from response: '{}'",
+                info.result.ledger.ledger_index
+            );
+        });
+    let mut rpc_ledger_hash = [0u8; 32];
+    hex::decode_to_slice(&info.result.ledger.ledger_hash, &mut rpc_ledger_hash[..])
+        .expect("unable to decode ledger hash");
+
+    // Wait for TmStatusChange message.
+    let check = |m: &BinaryMessage| {
+        matches!(
+            &m.payload,
+            Payload::TmStatusChange(TmStatusChange {
+                ledger_seq: Some(ledger_seq),
+                ledger_hash: Some(ledger_hash),
+                ..
+            })
+            if ledger_seq == &rpc_ledger_index && ledger_hash.as_slice() == rpc_ledger_hash
```

Comment:
```suggestion
            ledger_seq == &rpc_ledger_index && ledger_hash.as_slice() == rpc_ledger_hash
```
do double-check that this test case compiles, this doesn't seem right